src/In.ProjectEKA.HipService/Discovery/Patient/PatientDiscovery.cs

```diff
@@ -11,23 +11,28 @@ namespace In.ProjectEKA.HipService.Discovery.Patient
     public class PatientDiscovery : IDiscovery
     {
         private readonly Filter filter;
-        private readonly IMatchingRepository repo;
+        private readonly IMatchingRepository matchingRepository;
+        private readonly IDiscoveryRequestRepository discoveryRequestRepository;
 
-        public PatientDiscovery(IMatchingRepository patientRepository)
+        public PatientDiscovery(IMatchingRepository patientRepository,
+            IDiscoveryRequestRepository discoveryRequestRepository)
         {
-            repo = patientRepository;
+            matchingRepository = patientRepository;
+            this.discoveryRequestRepository = discoveryRequestRepository;
             filter = new Filter();
         }
 
         public async Task<Tuple<DiscoveryResponse, ErrorResponse>> PatientFor(DiscoveryRequest request)
         {
             var expression = GetExpression(request.Patient.VerifiedIdentifiers);
-            var patientInfos = await repo.Where(expression);
+            var patientInfos = await matchingRepository.Where(expression);
             var (patient, error) = DiscoveryUseCase.DiscoverPatient(filter.Do(patientInfos, request).AsQueryable());
 
-            return new Tuple<DiscoveryResponse, ErrorResponse>(
-                new DiscoveryResponse(patient),
-                error);
+            if (patient == null) return new Tuple<DiscoveryResponse, ErrorResponse>(null, error);

```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/gh/ProjectEKA/hip-service/pullRequest?prid=4833676)